symfony/client/src/orangehrmAdminPlugin/pages/organizationGeneralInformation/ViewOrganizationGeneralInformation.vue

```diff
@@ -0,0 +1,374 @@
+    </div>
+  </div>
+</template>
+
+<script>
+import {APIService} from '@orangehrm/core/util/services/api.service';
+import SwitchInput from "@orangehrm/oxd/src/core/components/Input/SwitchInput";
+
+export default {
+  props: {
+    numberOfEmployees: {
+      type: Number,
+      required: true,
+    },
+    countryList: {
+      type: Array,
+      required: true,
+    },
+  },
+  setup() {
+    const http = new APIService(
+      window.appGlobal.baseUrl,
+      'api/v2/admin/organization',
+    );
+    return {
+      http,
+    };
+  },
+
+  components: {
+    'oxd-switch-input': SwitchInput,
+  },
+
+  data() {
+    return {
+      editable: false,
+      isLoading: false,
+      organization: {
+        name: '',
+        noOfEmployees: this.numberOfEmployees,
+        registrationNumber: '',
+        taxId: '',
+        phone: '',
+        fax: '',
+        email: '',
+        street1: '',
+        street2: '',
+        city: '',
+        province: '',
+        country: '',
+        zipCode: '',
+        note: '',
+      },
+      rules: {
+        name: [],
+        noOfEmployees: [],
+        registrationNumber: [],
+        taxId: [],
+        phone: [],
+        fax: [],
+        email: [],
+        street1: [],
+        street2: [],
+        city: [],
+        province: [],
+        country: [],
+        zipCode: [],
+        note: [],
+      },
+      errors: [],
+    };
+  },
+
+  methods: {
+    onSave() {
+      this.isLoading = true;
+      this.http
+        .create({
+          name: this.organization.name,
+          registrationNumber: this.organization.registrationNumber,
+          taxId: this.organization.taxId,
+          phone: this.organization.phone,
+          fax: this.organization.fax,
+          email: this.organization.email,
+          street1: this.organization.street1,
+          street2: this.organization.street2,
+          city: this.organization.city,
+          province: this.organization.province,
+          country: this.organization.country[0].id,
+          zipCode: this.organization.zipCode,
+          note: this.organization.note,
+        })
+        .then(() => {
+          return this.$toast.success({
+            title: 'Success',
+            message: 'General Information updated successfully!',
+          });
+        })
+        .then(() => {
+          this.isLoading = false;
+          this.editable = false;
+        });
+    },
+  },
+  created() {
+    this.rules.name.push(v => {
+      return (!!v && v.trim() !== '') || 'Required';
+    });
+    this.rules.name.push(v => {
+      return (v && v.length <= 100) || 'Should not exceed 100 characters';
+    });
+    this.rules.registrationNumber.push(v => {
+      return v.length <= 30 || 'Should not exceed 30 characters';
+    });
+    this.rules.taxId.push(v => {
+      return v.length <= 30 || 'Should not exceed 30 characters';
+    });
+    this.rules.phone.push(v => {
+      return v.length <= 30 || 'Should not exceed 30 characters';
```

Comment:
check phone and fax numbers are valid phone numbers.

Reply:
added